azurelinuxagent/common/rdma.py

```diff
@@ -199,10 +199,24 @@ def start(self):
     def process(self):
         try:
             RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
-            RDMADeviceHandler.wait_rdma_device(
-                self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
-            RDMADeviceHandler.write_rdma_config_to_device(
-                self.rdma_dev, self.ipv4_addr, self.mac_addr)
+
+            skip_rdma_device = False
+            retcode,out = shellutil.run_get_output("modinfo hv_network_direct")
+            if retcode == 0:
```

Comment:
if retcode != 0, we just go through the normal behavior of writing to device file. The code will do the same thing before. Not sure if we want to give up and throw an exception here? In theory this modinfo command should never fail, unless something really bad happened on the system.